solutions/cses-1091.mdx

```diff
@@ -2,22 +2,25 @@
 id: cses-1091
 source: CSES
 title: Concert Tickets
-author: Danh Ta Chi Thanh
+author: Danh Ta Chi Thanh, Ben Dodge
 ---
 
-## Solution
+[Unofficial Editorial (C++)](https://codeforces.com/blog/entry/83295)
 
-[Unofficial Editorial](https://codeforces.com/blog/entry/83295)
 
-Another way of solving the problem is to use the `multiset::upper_bound`
-function to find the lowest-priced ticket that exceeds the maximum price for the
-current customer. When we have found the iterator pointing to the upper bound,
-we just decrement the iterator to get the maximum price possible for that
-current customer. Note that if the initial iterator points to the beginning of
-the multiset, we can infer that there is no ticket that works for the customer.
+## Explanation
 
-([Proof for my solution](https://stackoverflow.com/questions/9989731/algorithm-function-for-finding-last-item-less-than-or-equal-to-like-lower-bou/9989773#9989773))
+For each customer, we want to find the maximum possible ticket price they will
+accept from the available tickets. If the customer's maximum price is too low,
+so that no ticket price is below it, we can return -1. If not, we find the
+greatest ticket price less than or equal to the maximum price, and then remove
+that price as specified by the problem. We use a multiset for this.
 
+## Implementation 
+**Time Complexity:** $\mathcal{O}(n+ m)$
```

Comment:
I'm not sure about `NavigableMap`, but I think it has a log factor.

Reply:
You're right.